recipes/libunwind/all/conanfile.py

```diff
@@ -107,26 +107,26 @@ def package(self):
         shutil.copy(source_path, symlink_path)
 
     def package_info(self):
-        self.cpp_info.components["unwind"].names["pkg_config"] = "libunwind"
+        self.cpp_info.components["unwind"].set_property("pkg_config_name", "libunwind")
```

Comment:
I strongly advice to override global `pkg_config_name` property with something like `libunwind-do-not-use`, since this unwind component steals the default pkg_config_name `libunwind`.

Reply:
Hi @SpaceIm, thanks for the comment. I'm trying to understand this. If we change the global `pkg_config_name` wouldn't the situation be the same as if we don't? The component file always has preference over the global one so the component would be propagated downstream if we require libunwind independent from the global `pkg_config_name`. Am I missing something? Thanks a lot!

Reply:
the problem is to not propagate all components if downstream requires libunwind globally, it's a known issue of PkgConfigDeps & CMakeDeps when a specific component overrides pkg_config_name or cmake_target_name with the same name than root cpp_info.

Reply:
Yes, it should always take the component pc if it steals the name and never use an aggregated pc file, that behaviour is [documented since 1.45](https://docs.conan.io/en/latest/reference/conanfile/tools/gnu/pkgconfigdeps.html#names-and-aliases), so I think we should be fine

Reply:
well the issue I'm talking about is a little bit different, but I think it has been fixed by https://github.com/conan-io/conan/pull/11955

Reply:
Only to clarify the current behavior. If I, as consumer, have the following conanfile.py:

```python
        from conan import ConanFile

        class PkgConfigConan(ConanFile):
            name = "demo"
            version = "1.0"
            generators = "PkgConfigDeps"

            def requirements(self):
                self.requires("libunwind/1.0")

            def package_info(self):
                self.cpp_info.requires = ["libunwind::libunwind"]
```
Then, if I execute the command `install . -pr:h default -pr:b default`, the PC files resulting will be:

```
['libunwind-coredump.pc', 'libunwind-generic.pc', 'libunwind-ptrace.pc', 'libunwind-setjmp.pc', 'libunwind.pc']
```
Where that `libunwind.pc` could contain for instance:
```
prefix=/private/var/folders/s8/zgn806y90xq7bdh64hhfw9th0000gr/T/tmp129avqueconans/path with spaces/.conan2/p/6bac2b5ccb2956dc/p
libdir1=${prefix}/lib
includedir1=${prefix}/include

Name: libunwind
Description: Conan component: libunwind-libunwind
Version: 1.0
Libs: -L"${libdir1}" 
Cflags: -I"${includedir1}" 
```

Reply:
Yep, but for me the question was more: what is the content of `demo.pc`? Does it properly depend on each individual libunwind components in its `requires` field?

Reply:
It seems that it does not:

```
prefix=/private/var/folders/s8/zgn806y90xq7bdh64hhfw9th0000gr/T/tmp7mg5yc5dconans/path with spaces/.conan2/p/178fb6040fd4330a/p
libdir1=${prefix}/lib
includedir1=${prefix}/include

Name: demo
Description: Conan package: demo
Version: 1.0
Libs: -L"${libdir1}"
Cflags: -I"${includedir1}"
Requires: libunwind
```
It only requires `libunwind` as it's the component name (same as the root one).

Reply:
ok thanks, so there is still this issue and this review is still relevant :)

Reply:
I think so 👍

Reply:
I understand more clearly the problem now. I'll discuss the problem with the team next week and see if we can find a solution that serves for these specific cases.